inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/runnable/DisplayMessageRunnable.kt

```diff
@@ -80,42 +81,68 @@ internal class DisplayMessageRunnable(
         }
     }
 
+    private var timer: Timer? = null
     private fun downloadImage(view: InAppMessageBaseView, message: Message) {
         val url = message.getMessagePayload()?.resource?.imageUrl
-        // hide campaign view (cannot use visibility since Glide callback will not work.)
-        // https://github.com/bumptech/glide/issues/618
         if (url != null) {
-            view.alpha = 0f
-            Glide.with(view).load(url).addListener(object : RequestListener<Drawable> {
-                override fun onLoadFailed(
-                    e: GlideException?,
-                    model: Any?,
-                    target: Target<Drawable>?,
-                    isFirstResource: Boolean
-                ): Boolean {
-                    Timber.tag(TAG).d("Downloading image failed")
-                    // When image can't be downloaded, remove campaign.
-                    DisplayManager.instance().removeMessage(hostActivity)
-                    return false
-                }
+            Glide.with(view).load(url).timeout(IMG_DOWNLOAD_TIMEOUT).into(
+                    object : ImageViewTarget<Drawable>(view.findViewById(R.id.message_image_view)) {
+                        override fun onLoadStarted(placeholder: Drawable?) {
+                            super.onLoadStarted(placeholder)
+                            // hide campaign view (cannot use visibility since Glide callback will not work.)
+                            // https://github.com/bumptech/glide/issues/618
+                            view.alpha = 0f
```

Comment:
Is `onLoadStarted` always called almost immediately after calling `Glide.load(url)`?
I'm wondering if there is a chance for view to flicker (because until this call the alpha is 1)

Reply:
Yes, `onLoadStarted` is called immediately by Glide. But yes, I moved the setting of opacity since we don't have control on that behavior.